### PR TITLE
Ceph: Fixing the create-external-cluster-resources.py tests

### PR DESCRIPTION
**Pull request by aruniiird**
Signed-off-by: Arun Kumar Mohan <arun.iiird@gmail.com>

**Description of your changes:**
Fix the broken tests

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **CommitLint Bot**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip tests]

### CODE REVIEW

**Comment by aruniiird** [comment]
Fixed the unittest issue... Tests can be run on any machine, even without a ceph-cluster or python rados module...
@leseb , can you please take a look...

**Comment by leseb** [comment]
I added the `[skip tests]` flag so we don't run the integration suite but the only the build.

**Comment by aruniiird** [comment]
> Please add the `python -m unittest --verbose` command before this line https://github.com/rook/rook/blob/master/cluster/olm/ceph/generate-rook-csv.sh#L121 so that the python script unit tests can run on each build.

Made the said changes...
@leseb , can you please take a look...

File: cluster/olm/ceph/generate-rook-csv.sh [hunk 1]
```diff
@@ -119,6 +121,11 @@ function generate_csv(){
     mv "$DEFAULT_CSV_FILE_NAME" "$DESIRED_CSV_FILE_NAME"
     "${YQ_CMD_MERGE_OVERWRITE[@]}" "$DESIRED_CSV_FILE_NAME" "$ASSEMBLE_FILE_COMMON"
     "${YQ_CMD_WRITE[@]}" "$DESIRED_CSV_FILE_NAME" metadata.annotations.externalClusterScript "$(base64 <$CEPH_EXTERNAL_SCRIPT_FILE)"
+    if [ -f "$CEPH_EXTERNAL_SCRIPT_FILE" ];then
+        pushd $(dirname "$CEPH_EXTERNAL_SCRIPT_FILE") 1>/dev/null
+        python -m unittest --verbose "$(basename "$CEPH_EXTERNAL_SCRIPT_FILE" |sed -n 's@\.py$@@gp')"
```
Comment (leseb) [inline cluster/olm/ceph/generate-rook-csv.sh:126]: We need to install some python package or just this though a container...

Something like this:

```
docker run --rm -it  --entrypoint=python3 -v "/home/leseb/go/src/github.com/rook/rook/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:/create-external-cluster-resources.py:z" python -m unittest --verbose /create-external-cluster-resources.py
```

Comment (aruniiird) [reply cluster/olm/ceph/generate-rook-csv.sh:126]: We don't have to install anything, it uses the general `unittest` module, which comes with python (3 or 2) package...
The only criteria is to run on the same directory as the python file...

Comment (leseb) [reply cluster/olm/ceph/generate-rook-csv.sh:126]: The CI fails with: `/usr/bin/python: No module named unittest` so I assume there is something missing on that machine.

Comment (aruniiird) [reply cluster/olm/ceph/generate-rook-csv.sh:126]: Just an update,
I tried to install `pytest` (on my local machine) using pip3
`pip3 install pytest`
and ran the tests using following command
```
$ python -m pytest --verbose create-external-cluster-resources.py

============================================================================ test session starts =============================================================================
platform linux -- Python 3.7.7, pytest-5.4.2, py-1.8.1, pluggy-0.13.1 -- /home/testUser1/aruns/programming/python/my_env/bin/python
cachedir: .pytest_cache
rootdir: /home/testUser1/aruns/programming/python/my_env
collected 7 items                                                                                                                                                            

create-external-cluster-resources.py::TestRadosJSON::test_method_create_cephCSIKeyring_cephFSProvisioner PASSED                                                        [ 14%]
create-external-cluster-resources.py::TestRadosJSON::test_method_main_output PASSED                                                                                    [ 28%]
create-external-cluster-resources.py::TestRadosJSON::test_multi_data_pool_scenario PASSED                                                                              [ 42%]
create-external-cluster-resources.py::TestRadosJSON::test_multi_filesystem_scenario PASSED                                                                             [ 57%]
create-external-cluster-resources.py::TestRadosJSON::test_non_zero_return_and_error PASSED                                                                             [ 71%]
create-external-cluster-resources.py::TestRadosJSON::test_rbd_data_pool_name_arg PASSED                                                                                [ 85%]
create-external-cluster-resources.py::TestRadosJSON::test_valid_rgw_endpoint PASSED                                                                                    [100%]

============================================================================= 7 passed in 0.02s ==============================================================================
```

PS: using `pytest` you can run the test from any directory,
`python3 -m pytest --verbose my/test/env/newdir/create-external-cluster-resources.py`

**Comment by leseb** [comment]
Closing this due to inactivity. Feel free to re-open. Thanks.

**Comment by leseb** [comment]
@aruniiird let's go back to this once you have more time

**Comment by travisn** [comment]
@aruniiird Is this still needed?

**Comment by aruniiird** [comment]
This is not needed. We have fixed the tests in another PR...

**Pull request closed without merging**